sunpy/net/jsoc/jsoc.py

```diff
@@ -246,31 +267,21 @@ def request_data(self, jsoc_response, **kwargs):
         requestIDs = []
         for block in jsoc_response.query_args:
             # Do a multi-request for each query block
-            responses.append(self._multi_request(**block))
-            for i, response in enumerate(responses[-1]):
-                if response.status_code != 200:
-                    warn_message = "Query {0} retuned code {1}"
-                    warnings.warn(
-                        Warning(warn_message.format(i, response.status_code)))
-                    responses.pop(i)
-                elif response.json()['status'] != 2:
-                    warn_message = "Query {0} retuned status {1} with error {2}"
-                    json_response = response.json()
-                    json_status = json_response['status']
-                    json_error = json_response['error']
-                    warnings.warn(Warning(warn_message.format(i, json_status,
-                                                              json_error)))
-                    responses[-1].pop(i)
-
-            # Extract the IDs from the JSON
-            requestIDs += [response.json()['requestid'] for response in responses[-1]]
-
-        if return_responses:
-            return responses
-
-        return requestIDs
-
-    def check_request(self, requestIDs):
+
+            ds = self._make_recordset(**block)
+            cd = drms.Client(email=block['notify'])
+            r = cd.export(ds, method='url', protocol='fits')
+            r.wait()
```

Comment:
Yes. In the old impementation, it submitted the request, and the process ran in the background. Here, it waits for it to finish and then exits from the function. Won't this be better?